Neos.Neos/Classes/Domain/Pruning/RoleAndMetadataPruningProcessorFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Neos.Neos package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Domain\Pruning;
+
+use Neos\ContentRepository\Core\Factory\ContentRepositoryServiceFactoryDependencies;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryServiceFactoryInterface;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryServiceInterface;
+use Neos\Neos\Domain\Service\WorkspaceService;
+
+/**
+ * @implements ContentRepositoryServiceFactoryInterface<RoleAndMetadataPruningProcessor>
+ */
+final readonly class RoleAndMetadataPruningProcessorFactory implements ContentRepositoryServiceFactoryInterface
```

Comment:
i think this should not be a low level CR service

Reply:
Where should this be handled then. Also this is in the Neos.Neos Package for exactly that reason as it is needed but not part of the cr itself.

Reply:
i replaced it with a simple `new RoleAndMetadataPruningProcessor` ... we dont need the `ContentRepositoryServiceFactoryInterface` here for instantiation